src/v/io/page.h

```diff
@@ -88,6 +89,23 @@ class page {
      */
     void clear();
 
+    /*
+     * Used by the cache to test if this page may be evicted. If true, the cache
+     * is permitted to release the backing memory for this page and remove the
+     * page from the cache.
+     *
+     * use_count(): we maintain the invariant that a page in the cache is always
+     * referenced by an index structure. therefore, if its use_count is 1 then
+     * there are no other active references to the page, and the backing memory
+     * can be released. this property is useful because it bounds the set of
+     * locations where a page must be tested to those where new page references
+     * are obtained from an index.
+     */
+    [[nodiscard]] bool may_evict() const {
```

Comment:
fundamentally, no. a reader could cause start a page fault and then leave (e.g. cancel the read / timeout) while the background hydration is occurring so that the cache was the only reference to the page. in reality the io-queue holds a reference too, physically in the current implementation the implication you describe is present.